src/pages/ShowResource.tsx

```diff
@@ -18,46 +20,71 @@ function ShowResource({ location }: RouteComponentProps) {
 
   const { data: entity, isLoading, error, reload } = useAsync(
     getComitResourceFn,
-    { resourcePath, watch: location.pathname }
+    {
+      resourcePath,
+      watch: location.pathname
+    }
   );
+  const axiosError = error as AxiosError;
 
-  if (isLoading) {
-    return <CenteredProgress title="Fetching..." />;
-  } else if (
-    !error &&
-    entity &&
-    entity.class &&
-    entity.class.includes("swaps")
-  ) {
-    return (
+  const [showLoading, setShowLoading] = useState(false);
+  const [preventReload, setPreventReload] = useState(false);
+
+  useEffect(() => {
+    if (isLoading) {
+      setShowLoading(true);
+    }
+  }, [isLoading]);
+
+  useInterval(() => reload(), preventReload || isLoading ? null : 15000);
+
+  let resource;
```

Comment:
:wrench: Instead of re-assinging this variable in every branch, you could create a function that early returns in every branch.

The reason why I think that is better is because you don't have to keep the state in your mind about what is happening with this variable. Unless I am going through all the conditions in the branching below, I don't know upfront if it gets re-assigned several times based on the logic. By using a fn that early returns that is not possible at all (can only ever hit a single return path), hence when reading the code, I don't even have to think about that maybe happening and understand the code faster :)

Reply:
You are absolutely right.